Resources/Prototypes/Entities/Objects/Specific/nukedetector.yml

```diff
@@ -0,0 +1,54 @@
+- type: entity
+  parent: AnomalyLocatorUnpowered
+  id: NukeLocatorUnpowered
+  name: fissile material scanner
+  description: A device designed to scan for nuclear warheads in the local area. I sure hope it doesn't beep.
+  suffix: Unpowered
+  components:
+  - type: Sprite
+    sprite: Objects/Specific/Syndicate/nukelocator.rsi
+    layers:
+      - state: icon
+      - state: screen
+        shader: unshaded
+        visible: false
+        map: ["enum.PowerDeviceVisualLayers.Powered"]
+  - type: ProximityDetector
+    enabled: false
+    range: 50
+    criteria:
+      components:
+      - NukeLabel # NukeLabel catches the decoy nuke too, which is kinda funny.
```

Comment:
Have you profiled this because proximitydetector is extremely slow and you've given it the range of the station

Reply:
With the caveat I don't really know what I'm doing, I tried Visual Studios performance profiler and spawned 13 on Oasis without seeing much difference.

Reply:
if you absolutely have to do this you should EntityQueryNumerator<NukeLabel>() and iterate that it should be faster